ext/bg/js/dictionary.js

```diff
@@ -132,26 +132,27 @@ function dictTermsGroup(definitions, dictionaries) {
         }
 
         const keyString = key.toString();
-        if (hasOwn(groups, keyString)) {
-            groups[keyString].push(definition);
-        } else {
-            groups[keyString] = [definition];
+        let groupDefinitions = groups.get(keyString);
+        if (typeof groupDefinitions === 'undefined') {
+            groupDefinitions = [];
+            groups.set(keyString, groupDefinitions);
         }
+
+        groupDefinitions.push(definition);
     }
 
     const results = [];
-    for (const key in groups) {
-        const groupDefs = groups[key];
-        const firstDef = groupDefs[0];
-        dictTermsSort(groupDefs, dictionaries);
+    for (const groupDefinitions of groups.values()) {
+        const firstDef = groupDefinitions[0];
+        dictTermsSort(groupDefinitions, dictionaries);
         results.push({
-            definitions: groupDefs,
+            definitions: groupDefinitions,
             expression: firstDef.expression,
             reading: firstDef.reading,
             furiganaSegments: firstDef.furiganaSegments,
             reasons: firstDef.reasons,
             termTags: firstDef.termTags,
-            score: groupDefs.reduce((p, v) => v.score > p ? v.score : p, Number.MIN_SAFE_INTEGER),
+            score: groupDefinitions.reduce((p, v) => v.score > p ? v.score : p, Number.MIN_SAFE_INTEGER),
```

Comment:
I don't remember any more why this `Number.MIN_SAFE_INTEGER` is here, but you removed it from another place in an earlier commit: https://github.com/FooSoft/yomichan/commit/5541aae2012785894fbb52b2767cce5be9a6f5ba

Reply:
It's used as the default value in case the array is empty. It was removed in the reference commit since there was guaranteed to be a value.